content/06.know-your-problem.md

```diff
@@ -31,7 +31,12 @@ In general, deep learning has an increased tendency for overfitting, compared to
 For a large dataset, overfitting may not be a concern, but the modeling power of deep learning may lead to more spurious correlations and thus incorrect interpretation of results (see [Tip 9](#interpretation)).
 It is important to note that molecular or imaging datasets often require appropriate clinical or demographic data to support robust analyses; this must always be balanced with the need to protect patient privacy (see [Tip 10](#privacy)).
 Looking at these annotations can also clarify the study design (for example, by seeing if all the individuals are adolescents or women) or at least help the analyst employing deep learning to know what questions to ask.
+
 One tool in the practitioner's toolkit for dealing with datasets that are not thoroughly understood is data simulation.
 Simulating data under reasonable assumptions and assessing the extent to which the model can capture or mitigate issues with certain features of the data is a valuable technique for identifying model architectures and hyperparameters that are reasonable under those assumptions.
+To accurately test the performance of the model, it is important that simulated datasets be generated for a range of parameters.
+Simulated datasets generated from parameters that are consistent with the assumptions of the deep learning models can be used to verify the correctness of the model’s implementation.
+Further varying the parameters to violate the model's assumptions can test the sensitivity of the model's performance to those assumptions.
+Generative models based upon sample training sets can be an important tool for generating large cohorts of simulated datasets that are representative of the real world problems to which the machine learning prediction models will be applied.

```

Comment:
Can we clarify this sentence a bit? I'm still not 100% sure I understand what its saying. Does it mean that generative models are useful for testing predictive models (like what the last sentence said)? Is it a conclusion? Or is it introducing a different idea which I failed to catch?

Reply:
I re-read. It is a bit confusing. Seems like it can either be the conclusion or the start of going in a more specific direction (since it seems to use the word generative for the first time). Not sure the last sentence is needed.

Reply:
Now that I re-read, I think this paragraph needs to be tweaked. The intro sentence is talking about datasets not being thoroughly understood...so meaning the true distribution is unknown? If the idea is to learn the true data distribution of the data for the purpose of generating new, simulated data, then I think this should be spelled out for the reader. Do we also want to spell out generative vs discriminative? GAN vs VAE?

Not sure if this is commit worthy, but for a starting place how about something like

> Data simulation is a powerful approach to creating additional data for model testing. In data simulation, a generative model is used to learn the true distribution of a training set for the purpose of generating new data points. Simulations should be performed under reasonable assumptions since the goal would be to identify useful model architectures and hyperparameters, and simulated datasets can be used to verify the correctness of a model’s implementation. To accurately test the performance of the model, it is important that simulated datasets be generated for a range of parameters. For example, varying the parameters to violate the model's assumptions can test the sensitivity of the model's performance.

Reply:
I remember @rasbt suggested we avoid specifically talking about VAEs due to their declining use (correct me if I'm wrong). Your paragraph is clearer so I suggest we use that in its place since it covers the same topics as the current suggested change. If someone else agrees, I'll update the PR and re-request reviews.

Reply:
I think it depends a bit. We used VAEs recently to better understand a phenomenon that had been widely observed but not really empirically demonstrated: https://www.biorxiv.org/content/10.1101/2020.05.03.066597v1 . The data from the VAE are clearly a bit over-smoothed, but depending on your use case that might or might not matter.

There are also elements of data simulation that aren't based on a generative neural network (i.e., learning the true distribution of a training set) and are instead based upon researcher suppositions about the problem at hand. If we want to start from @pstew's I'm happy to suggest some modifications on top of it! 👍

Reply:
I guess we could also see if @ajlee21 wants to weigh in on revising the paragraph from @pstew since she led that dataset simulation work.

Reply:
Sure, very happy to help with this. This manuscript sounds like it will be very helpful!. Once @pstew 's changes are merged in I'm happy to add to this

Reply:
@pstew - can you go to files changed and mark as approve? then we can merge this and @ajlee21 can contribute a PR with suggested modifications

Reply:
@ajlee21 Thanks! I made the changes. Please tweak as much as needed. Since we took this from another rule, do we need to ultimately bring this paragraph back around to the idea of this rule (knowing your data/knowing your problem)? I suppose getting to the true distribution of the data is a start, but maybe as written it doesn't exactly fit.

Reply:
@cgreene Done!